src/components/ChannelPreview/hooks/useChannelPreviewDisplayName.js

```diff
@@ -0,0 +1,26 @@
+import { useState, useEffect, useContext } from 'react';
+import { ChatContext } from '../../../context';
+
+export const useChannelPreviewDisplayName = (channel) => {
+  const { client } = useContext(ChatContext);
+  const [displayName, setDisplayName] = useState('');
+
+  useEffect(() => {
+    if (channel.data.name && typeof channel.data.name === 'string') {
```

Comment:
optional chaining can help here
```suggestion
    if (typeof channel?.data?.name === 'string') {
```